src/org/sosy_lab/java_smt/solvers/cvc4/CVC4TheoremProver.java

```diff
@@ -202,11 +203,12 @@ public boolean isUnsat() throws InterruptedException, SolverException {
     }
 
     Result result;
-    try (ShutdownHook hook = new ShutdownHook(shutdownNotifier, smtEngine::interrupt)) {
-      shutdownNotifier.shutdownIfNecessary();
+    try (ShutdownHook hook =
+        new ShutdownHook(proverShutdownManager.getNotifier(), smtEngine::interrupt)) {
+      proverShutdownManager.getNotifier().shutdownIfNecessary();
       result = smtEngine.checkSat();
     }
-    shutdownNotifier.shutdownIfNecessary();
+    proverShutdownManager.getNotifier().shutdownIfNecessary();
```

Comment:
I would prefer to avoid calling `getNotifier()` more than once here.

Reply:
I just forgot to push everything. Should be removed now. Sorry for the delay.